compile pedis error

[3/4] CXX build/release/commit_log.o
FAILED: build/release/commit_log.o 
/opt/scylladb/bin/g++ -MD -MT build/release/commit_log.o -MF build/release/commit_log.o.d -std=gnu++17 -g  -Wall -Werror -Wno-error=deprecated-declarations -fvisibility=hidden -Wno-attributes -pthread -I/u01/jeff.dk/github/pedis/seastar -U_FORTIFY_SOURCE  -march=nehalem -fconcepts -I/u01/jeff.dk/github/pedis/seastar/fmt/include -DBOOST_TEST_DYN_LINK -Wno-ignored-attributes -Wno-overloaded-virtual -Wno-maybe-uninitialized -Wno-error=cpp -Wno-stringop-overflow -DSEASTAR_HAVE_GCC6_CONCEPTS -DSEASTAR_NO_EXCEPTION_HACK -DNO_EXCEPTION_INTERCEPT -DSEASTAR_HAVE_HWLOC -DSEASTAR_HAVE_NUMA -DSEASTAR_HAVE_LZ4_COMPRESS_DEFAULT -O2  -I/u01/jeff.dk/github/pedis/seastar/build/release/gen -I/u01/jeff.dk/github/pedis/seastar/build/release/c-ares -I/usr/include/jsoncpp -DBOOST_TEST_DYN_LINK  -Wno-maybe-uninitialized -Wno-tautological-compare -Wno-missing-braces -Wno-error=deprecated-declarations -DHAVE_LIBSYSTEMD=1 -I. -I build/release/gen -I seastar -I seastar/build/release/gen -c -o build/release/commit_log.o commit_log.cc
commit_log.cc: In function ‘void store::init_type_crc(uint32_t*)’:
commit_log.cc:98:21: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
   for (int i = 0; i <= MAX_RECORD_TYPE; i++) {
                   ~~^~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
ninja: build stopped: subcommand failed

Looks like a minor bug to me, patch is welcomed


On Tue, Nov 6, 2018 at 10:46 PM Kai Ding <notifications@github.com> wrote:

> [3/4] CXX build/release/commit_log.o
> FAILED: build/release/commit_log.o
> /opt/scylladb/bin/g++ -MD -MT build/release/commit_log.o -MF
> build/release/commit_log.o.d -std=gnu++17 -g -Wall -Werror
> -Wno-error=deprecated-declarations -fvisibility=hidden -Wno-attributes
> -pthread -I/u01/jeff.dk/github/pedis/seastar -U_FORTIFY_SOURCE
> -march=nehalem -fconcepts -I/u01/jeff.dk/github/pedis/seastar/fmt/include
> -DBOOST_TEST_DYN_LINK -Wno-ignored-attributes -Wno-overloaded-virtual
> -Wno-maybe-uninitialized -Wno-error=cpp -Wno-stringop-overflow
> -DSEASTAR_HAVE_GCC6_CONCEPTS -DSEASTAR_NO_EXCEPTION_HACK
> -DNO_EXCEPTION_INTERCEPT -DSEASTAR_HAVE_HWLOC -DSEASTAR_HAVE_NUMA
> -DSEASTAR_HAVE_LZ4_COMPRESS_DEFAULT -O2 -I/u01/
> jeff.dk/github/pedis/seastar/build/release/gen -I/u01/
> jeff.dk/github/pedis/seastar/build/release/c-ares -I/usr/include/jsoncpp
> -DBOOST_TEST_DYN_LINK -Wno-maybe-uninitialized -Wno-tautological-compare
> -Wno-missing-braces -Wno-error=deprecated-declarations -DHAVE_LIBSYSTEMD=1
> -I. -I build/release/gen -I seastar -I seastar/build/release/gen -c -o
> build/release/commit_log.o commit_log.cc
> commit_log.cc: In function ‘void store::init_type_crc(uint32_t*)’:
> commit_log.cc:98:21: error: comparison between signed and unsigned integer
> expressions [-Werror=sign-compare]
> for (int i = 0; i <= MAX_RECORD_TYPE; i++) {
> ~~^~~~~~~~~~~~~~~~~~
> cc1plus: all warnings being treated as errors
> ninja: build stopped: subcommand failed
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/fastio/pedis/issues/26>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABp6RT55mlXyvtcWPOyaAMbcuUUPI4SHks5usoHVgaJpZM4YR3py>
> .
>

> Looks like a minor bug to me, patch is welcomed
> […](#)
> On Tue, Nov 6, 2018 at 10:46 PM Kai Ding ***@***.***> wrote: [3/4] CXX build/release/commit_log.o FAILED: build/release/commit_log.o /opt/scylladb/bin/g++ -MD -MT build/release/commit_log.o -MF build/release/commit_log.o.d -std=gnu++17 -g -Wall -Werror -Wno-error=deprecated-declarations -fvisibility=hidden -Wno-attributes -pthread -I/u01/jeff.dk/github/pedis/seastar -U_FORTIFY_SOURCE -march=nehalem -fconcepts -I/u01/jeff.dk/github/pedis/seastar/fmt/include -DBOOST_TEST_DYN_LINK -Wno-ignored-attributes -Wno-overloaded-virtual -Wno-maybe-uninitialized -Wno-error=cpp -Wno-stringop-overflow -DSEASTAR_HAVE_GCC6_CONCEPTS -DSEASTAR_NO_EXCEPTION_HACK -DNO_EXCEPTION_INTERCEPT -DSEASTAR_HAVE_HWLOC -DSEASTAR_HAVE_NUMA -DSEASTAR_HAVE_LZ4_COMPRESS_DEFAULT -O2 -I/u01/ jeff.dk/github/pedis/seastar/build/release/gen -I/u01/ jeff.dk/github/pedis/seastar/build/release/c-ares -I/usr/include/jsoncpp -DBOOST_TEST_DYN_LINK -Wno-maybe-uninitialized -Wno-tautological-compare -Wno-missing-braces -Wno-error=deprecated-declarations -DHAVE_LIBSYSTEMD=1 -I. -I build/release/gen -I seastar -I seastar/build/release/gen -c -o build/release/commit_log.o commit_log.cc commit_log.cc: In function ‘void store::init_type_crc(uint32_t*)’: commit_log.cc:98:21: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare] for (int i = 0; i <= MAX_RECORD_TYPE; i++) { ~~^~~~~~~~~~~~~~~~~~ cc1plus: all warnings being treated as errors ninja: build stopped: subcommand failed — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub <#26>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ABp6RT55mlXyvtcWPOyaAMbcuUUPI4SHks5usoHVgaJpZM4YR3py> .

I will fix it later

> Looks like a minor bug to me, patch is welcomed
> […](#)
> On Tue, Nov 6, 2018 at 10:46 PM Kai Ding ***@***.***> wrote: [3/4] CXX build/release/commit_log.o FAILED: build/release/commit_log.o /opt/scylladb/bin/g++ -MD -MT build/release/commit_log.o -MF build/release/commit_log.o.d -std=gnu++17 -g -Wall -Werror -Wno-error=deprecated-declarations -fvisibility=hidden -Wno-attributes -pthread -I/u01/jeff.dk/github/pedis/seastar -U_FORTIFY_SOURCE -march=nehalem -fconcepts -I/u01/jeff.dk/github/pedis/seastar/fmt/include -DBOOST_TEST_DYN_LINK -Wno-ignored-attributes -Wno-overloaded-virtual -Wno-maybe-uninitialized -Wno-error=cpp -Wno-stringop-overflow -DSEASTAR_HAVE_GCC6_CONCEPTS -DSEASTAR_NO_EXCEPTION_HACK -DNO_EXCEPTION_INTERCEPT -DSEASTAR_HAVE_HWLOC -DSEASTAR_HAVE_NUMA -DSEASTAR_HAVE_LZ4_COMPRESS_DEFAULT -O2 -I/u01/ jeff.dk/github/pedis/seastar/build/release/gen -I/u01/ jeff.dk/github/pedis/seastar/build/release/c-ares -I/usr/include/jsoncpp -DBOOST_TEST_DYN_LINK -Wno-maybe-uninitialized -Wno-tautological-compare -Wno-missing-braces -Wno-error=deprecated-declarations -DHAVE_LIBSYSTEMD=1 -I. -I build/release/gen -I seastar -I seastar/build/release/gen -c -o build/release/commit_log.o commit_log.cc commit_log.cc: In function ‘void store::init_type_crc(uint32_t*)’: commit_log.cc:98:21: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare] for (int i = 0; i <= MAX_RECORD_TYPE; i++) { ~~^~~~~~~~~~~~~~~~~~ cc1plus: all warnings being treated as errors ninja: build stopped: subcommand failed — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub <#26>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ABp6RT55mlXyvtcWPOyaAMbcuUUPI4SHks5usoHVgaJpZM4YR3py> .

i have made a pr, see https://github.com/fastio/pedis/pull/27

--- a/.gitmodules
+++ b/.gitmodules
@@ -1,4 +1,4 @@
 [submodule "seastar"]
 	path = seastar
-	url = https://github.com/scylladb/scylla-seastar.git 
+	url = https://github.com/scylladb/seastar.git 
 	ignore = dirty

--- a/commit_log.cc
+++ b/commit_log.cc
@@ -95,7 +95,7 @@ public:
 };
 
 static void init_type_crc(uint32_t* type_crc) {
-  for (int i = 0; i <= MAX_RECORD_TYPE; i++) {
+  for (uint32_t i = 0; i <= MAX_RECORD_TYPE; i++) {
     char t = static_cast<char>(i);
     type_crc[i] = crc32c::value(&t, 1);
   }

--- a/configure.py
+++ b/configure.py
@@ -366,7 +366,7 @@ else:
 # a list element means a list of alternative packages to consider
 # the first element becomes the HAVE_pkg define
 # a string element is a package name with no alternatives
-optional_packages = [['libsystemd', 'libsystemd-daemon']]
+optional_packages = [['libsystemd', 'libsystemd-daemon', 'systemd']]
 pkgs = []
 
 def setup_first_pkg_of_list(pkglist):

